Tradução: EDA.qmd (#85)
cienciadedatos/pt-r4ds: EDA.qmd

# Análise Exploratória dos Dados {#sec-exploratory-data-analysis}

```{r}
#| echo: false

source("_common.R")
```

## Introdução

Esse capítulo mostrará a você como usar visualização e transformação para explorar seus dados em uma forma sistemática, uma tarefa que os estatísticos chamam de análise exploratória dos dados, ou EDA (Exploratory Data Analysis) abreviamente.
EDA é um ciclo iterativo.
Você:

1.  Gera questões sobre os seus dados.

2.  Encontra respostas visualizando, transformando e modelando seus dados.

3.  Usa o que aprendeu para refinar suas questões e/ou gerar novas questões.

A EDA não é um processo formal com um conjunto estrito de regras.
Mais que qualquer coisa, EDA é um estado da mente.
Durante as fases inicias da EDA você deverá se sentir livre para investigar cada ideia que ocorre a você.
Algumas dessas ideias darão certo e outras serão becos sem saída.

> À medida que sua exploração continua, você se concentrará em alguns insights particularmente produtivos que eventualmente escreverá e comunicará a outras pessoas.

Reviewer: ```suggestion
À medida que sua exploração continua, você se concentrará em alguns insights particularmente produtivos que você eventualmente compilará e comunicará a outras pessoas.
```

Eu acho que "compilar" apesar de ter um significado específico em programação, transmite melhor a idéia que ele quer passar nesse contexto: de pegar as informações que achar importantes para uma apresentação.

A EDA é uma parte importante de qualquer análise de dados, mesmo que as principais questões de pesquisa sejam entregues a você de bandeja, porque você sempre precisa investigar a qualidade dos seus dados.

> Limpeza de dados é apenas uma das aplicações da EDA: você se questiona sobre se seus dados vão de encontro às suas expectativas ou não.

Reviewer: ```suggestion
Limpeza de dados é apenas uma das aplicações da EDA: você questiona se seus dados estão de acordo com suas expectativas ou não.
```

> Para fazer limpeza de dados, você irá necessitar implementar todas as ferramentas da EDA: visualização, transformação e modelagem.

Reviewer: ```suggestion
Para fazer limpeza de dados, você precisará implementar todas as ferramentas da EDA: visualização, transformação e modelagem.
```

### Pré-requisitos

> Nesse capítulo, nós iremos combinar o que você tem aprendido sobre dplyr e ggplot2 para fazer perguntas interativamente, responder com dados e, em seguida, fazer novas perguntas.

Reviewer 2: ```suggestion
Nesse capítulo, combinaremos o que você tem aprendido sobre dplyr e ggplot2 para fazer perguntas interativamente, responder com dados e, em seguida, fazer novas perguntas.
```

Isso é verdade mesmo se você mede quantidades constantes, como a velocidade da luz.
Cada uma de suas medidas irá incluir uma pequena quantidade de erro que varia de medida a medida.
Variáveis também muda se você mede diferentes objetos (por exemplo, cor dos olhos de diferentes pessoas) ou diferentes tempos (por exemplo, os níveis de energia de um elétron em diferentes momentos).
Cada variável tem seu próprio padrão de variação, que pode revelar interessantes informações sobre como isso varia entre medições na mesma observação, bem como entre observações.
A melhor forma de entender o padrão é visualizando a distribuição dos valores da variável, que você tem aprendido sobre no @sec-data-visualization.

Vamos iniciar nossa exploração visualizando a variação nos pesos (`quilate`) de \~54,000 diamantes do `diamante`.
Como `quilate` é uma variável numérica, podemos usar um histograma:

```{r}
#| fig-alt: |
#| Um histograma de quilates de diamantes, com o eixo x variando de 0 a 4,5
#| e o eixo y variando de 0 a 30.000. A distribuição é distorcida para a direita

> #| com muito poucos diamantes na caixa centralizada em 0, quase 30.000 diamantes na

Reviewer 2: Classe realmente é o que faz mais sentido

#| caixa centralizada em 0,5, aproximadamente 15.000 diamantes na caixa centralizada
#| em 1, e muito menos, aproximadamente 5.000 diamantes na caixa centralizada em

> #| 1.5. Além disso, há uma cauda arrastada.

Reviewer 2: Também pensei em cauda pesada, mas concordo em fazer o texto ser compreendido por mais pessoas.

library(dados)

ggplot(diamante, aes(x = quilate)) +
  geom_histogram(binwidth = 0.5)
```

Agora que você pode visualizar a variação, o que você deve observar em seus gráficos?
E que tipo de perguntas de acompanhamento você deve fazer?
Nós temos colocado abaixo uma lista dos tipos mais comuns de informações que você encontrará em seus gráficos, junto com algumas perguntas de acompanhamento para cada tipo de informação.

> A chave para fazer boas questões de acompanhamento irá depender da curiosidade (o que mais você quer aprender?) bem como do seu ceticismo (Como isso pode ser enganoso?).

Reviewer 2: ```suggestion
A chave para fazer boas questões de acompanhamento será contar com a sua curiosidade (o que mais você quer aprender?) bem como do seu ceticismo (Como isso pode ser enganoso?).
```

### Valores típicos

Em ambos gráficos de barras e histogramas, barras altas mostram valores comuns de uma variável, e barras mais curtas mostram valores menos comuns.

> Locais que não tem barras revelam valores que não foram vistos nos seus dados.

Reviewer 2: ```suggestion
Locais que não têm barras revelam valores que não foram vistos nos seus dados.
```

Para transformar essas informações em perguntas úteis, observe qualquer coisa não esperada:

-   Quais valores são os mais comuns?
    Por quê?

-   Quais valores são raros?
    Por quê?
    Isso corresponde às suas expectativas?

-   Você consegue ver padrões incomuns?
    O que pode explicar eles?

Vamos observar a distribuição de `quilate` para os diamantes menores.

```{r}
#| fig-alt: |
#| Um histograma de quilates de diamantes, com o eixo x variando de 0 a 3 e

> #| o eixo y variando de 0 a aproximadamente 2.500. A largura do compartimento é 

Reviewer 2: Apesar de ser uma linguagem mais específica acho que vale a pena sim mudar para amplitude da classe. A ideia do livro é também trazer pessoas para a comunidade, o que envolve familiarizar as pessoas com a terminologia.

#| bastante estreita (0,01), resultando em um número muito grande de barras finas. 
#| A distribuição é inclinada para a direita, com muitos picos seguidos por barras em
#| alturas decrescentes, até um aumento acentuado no próximo pico.

menores_diamantes <- diamante |> 
  filter(quilate < 3)

ggplot(menores_diamantes, aes(x = quilate)) +
  geom_histogram(binwidth = 0.01)
```

Esse histograma sugere várias questões interessantes:

-   Por que há mais diamantes em números inteiros de quilates e frações comuns de quilates?

-   Por que há mais diamantes ligeiramente à direita de cada pico do que há ligeiramente à esquerda de cada pico?

Visualizações também podem revelar agrupamentos, os quais sugerem que existem subgrupos em seus dados.
Para entender os subgrupos, questione:

> -   Como são as observações dentro de cada subgrupo similares uns aos outros?

Reviewer 2: ```suggestion
-   Quão similares as observações dentro de cada subgrupo são umas às outras?
```

É o mesmo da Carol, só estou sugerindo remover o "O".

> -   Como são as observações em grupos diferentes separados uns dos outros?

Reviewer 2: ```suggestion
-   Quão diferentes as observações em grupos separados são umas das outras?
```

Segue a mesma sugestão, remover o "O" inicial

> -   Como são as observações em grupos diferentes separados uns dos outros?

Reviewer: acho q eu inventei esse artigo ai na frente do quão ne

ggplot(menores_diamantes, aes(x = quilate, y = preco)) +
  geom_hex()
```

Outra opção é discretizar uma variável contínua para que ela atue como uma variável categórica.
Assim você consegue usar uma das técnicas para visualizar a combinação de uma variável categórica e uma contínua que você aprendeu antes.
Por exemplo, você poderia categorizar a variável `quilate`, e então para cada grupo exibir um boxplot:

```{r}
#| fig-alt: |
#| Gráficos boxplot lado a lado de preços por quilate. Cada boxplot representa
#| diamantes que têm 0,1 quilates de diferença em peso. Os boxplots mostram
#| que à medida que quilate aumenta ocorrem aumentos do preço médio. Além disso,
#| diamantes com 1,5 quilates ou menos têm distribuições de preços assimétricos à
#| direita, 1,5 a 2 têm distribuições de preços aproximadamente simétricas e os
#| diamantes que pesam mais têm distribuições assimétricas à esquerda. Diamantes menores
#| e mais baratos têm valores discrepantes na extremidade superior, diamantes maiores,
#| e mais caros têm valores discrepantes na extremidade inferior.

ggplot(menores_diamantes, aes(x = quilate, y = preco)) + 
  geom_boxplot(aes(group = cut_width(quilate, 0.1)))
```

> `cut_width(x, width)`, as used above, divides `x` into bins of width `width`.
> By default, boxplots look roughly the same (apart from number of outliers) regardless of how many observations there are, so it's difficult to tell that each boxplot summaries a different number of points.

Reviewer 3: ```suggestion

```

> One way to show that is to make the width of the boxplot proportional to the number of points with `varwidth = TRUE`.

Reviewer 3: ```suggestion

```

options(old)
```

A variável `y` mede uma das três dimensões desses diamantes, em mm.

> Sabemos que os diamantes não podem ter largura de 0 mm, então esses valores devem estar incorretos.

Reviewer 2: ```suggestion
Sabemos que os diamantes não podem ter largura de 0mm, então esses valores devem estar incorretos.
```

> Ao fazer EDA, descobrimos dados faltantes codificados como 0, que nunca teríamos encontrado simplesmente procurando por `NA`s.do a diantes, nós podemos escolher recodificar estes valores como `NA`s, a fim de evitar cálculos enganosos.

Reviewer 2: ```suggestion
Ao fazer EDA, descobrimos dados faltantes codificados como 0, que nunca teríamos encontrado simplesmente procurando por `NA`s.
Em seguida, podemos escolher recodificar estes valores como `NA`s, a fim de evitar cálculos enganosos.
```

> Também podemos suspeitar que as medidas de 32 mm e 59 mm são implausíveis: esses diamantes têm mais de uma polegada de comprimento, mas não custam centenas de milhares de dólares!

Reviewer 2: ```suggestion
Também podemos suspeitar que as medidas de 32mm e 59mm são implausíveis: esses diamantes têm mais de uma polegada de comprimento, mas não custam centenas de milhares de dólares!
```

    (Dica: pense cuidadosamente sobre o `binwidth` e tenha certeza que você tentou uma ampla gama de valores.)

3.  Quantos diamantes equivalem a 0,99 quilates?
    Quantos equivalem a 1 quilate?
    O que você acha que é a causa da diferença?

4.  Compare e contraste `coord_cartesian()` vs. `xlim()` ou `ylim()` quando ampliar em um histograma.
    O que acontece se você deixa o `binwidth` desativado?
    O que acontece se você tentar ampliar de forma que apenas meia barra seja exibida?

## Lidando com valores atípicos {#sec-unusual-values-eda}

Se você encontrou valores atípicos no seu conjunto de dados e simplesmente deseja prosseguir com o restante das suas análises, você tem duas opções:

1.  Eliminar a linha inteira com os valores estranhos:

    ```{r}
    #| eval: false

    diamante2 <- diamante |> 
      filter(between(y, 3, 20))
    ```

    Não recomendamos esta opção porque um valor inválido não implica que todos os outros valores para essa observação também sejam inválidos.
    Além disso, se você tiver dados de baixa qualidade, ao aplicar essa abordagem a todas as variáveis, você poderá descobrir que não tem mais dados!

> 2.  Ao invés, nós recomendamos substituir os valores atípicos por valores faltantes (missing values).O caminho mais fácil para fazer isso, é usar a função mutate() para substituir os valores atípicos da variável.Você também pode usar o if_else() para substituir os valores atípicos por NA:

Reviewer: ```suggestion
2.  Ao invés disso, nós recomendamos substituir os valores atípicos por valores faltantes (missing values). O caminho mais fácil para fazer isso, é usar a função mutate() para substituir os valores atípicos da variável. 
Você também pode usar o if_else() para substituir os valores atípicos por NA:
```

Reviewer 2: @cmusso86 pode editar essa sugestão para dividir em paragrafos diferentes, conforme o original?

#|   has length greater than 3. The one outlier has a length of 0 and a width 
#|   of about 6.5. 
#| Um gráfico de dispersão de larguras versus comprimentos de diamantes. Há um forte,
#| associação linear entre as duas variáveis. Todos menos um dos diamantes
#| tem comprimento maior que 3. O valor discrepante tem comprimento 0 e largura
#| de cerca de 6,5.

ggplot(diamonds2, aes(x = x, y = y)) + 
ggplot(diamante2, aes(x = x, y = y)) + 
  geom_point()
```

To suppress that warning, set `na.rm = TRUE`:
Para suprimir esse aviso, defina `na.rm = TRUE`:

```{r}
#| eval: false

ggplot(diamonds2, aes(x = x, y = y)) + 
ggplot(diamante2, aes(x = x, y = y)) + 
  geom_point(na.rm = TRUE)
```

Other times you want to understand what makes observations with missing values different to observations with recorded values.
For example, in `nycflights13::flights`[^eda-1], missing values in the `dep_time` variable indicate that the flight was cancelled.
So you might want to compare the scheduled departure times for cancelled and non-cancelled times.
You can do this by making a new variable, using `is.na()` to check if `dep_time` is missing.

> [^eda-1]: Remember that when we need to be explicit about where a function (or dataset) comes from, we'll use the special form `package::function()` or `package::dataset`.

Reviewer 2: Eu inseri tanto a tag da nota de rodapé quanto o próprio texto da nota nas miinhas sugestões.

> Por exemplo, em `nycflights13::flights1`, valores ausentes na variável dep_time indicam que o voo foi cancelado.

Reviewer 2: ```suggestion
Por exemplo, em `nycflights13::flights1`[^eda-1], valores ausentes na variável dep_time indicam que o voo foi cancelado.
```

> Você pode fazer isso criando uma nova variável, usando `is.na()` para verificar se `dep_time` está faltando.

Reviewer 2: ```suggestion
Você pode fazer isso criando uma nova variável, usando `is.na()` para verificar se `dep_time` está faltando.

[^eda-1]: Lembre-se que quando é necessário ser explícito sobre o local de origem da função (ou conjunto de dados), usaremos a forma especial `pacote::função()` ou `pacote::dados`.
```

>     O que acontece com valores perdidos em um gráfico de barras?

Reviewer: ```suggestion
    O que acontece com valores ausentes em um gráfico de barras?
```

>     Por que existe uma diferença em como valores perdidos são tratados em histogramas e gráficos de barras?

Reviewer: ```suggestion
    Por que existe uma diferença em como valores ausentes são tratados em histogramas e gráficos de barras?
```

> 3.  Recrie o gráfico de frequência de `saida_programada` colorido de acordo com o cancelamento ou não do voo.

Reviewer: ```suggestion
3.  Recrie o gráfico de frequência de `saida_programada` colorido de acordo com o voo ter sido cancelado ou não.
```

>     Também use o facetamento para a variável `cancelado` variable.

Reviewer: ```suggestion
    Também use o facetamento para a variável `cancelado`.
```

>     Experimente diferentes valores da variável `scales` na função de lapidação para mitigar o efeito de mais voos não cancelados do que voos cancelados.

Reviewer: ```suggestion
    Experimente diferentes valores do argumento `scales` na função de facetamento para mitigar o efeito de mais voos não cancelados do que voos cancelados.
```

a tradução estava certa, mas achei estranho chamarem o argumento da função de "variável" no original

Reviewer 2: De acordo.

> ## Covariação

Reviewer: continuarei daqui

Por exemplo, vamos explorar como o preço do diamante varia com a qualidade dele (medida pelo `corte`) usando `geom_frenqpoly()`:

```{r}
#| fig-alt: |
#| Um polígono de frequência de preços de diamantes onde cada corte de quilate (Justo,
#| Bom, Muito Bom, Premium e Ideal) é representado com uma cor diferente

> #| linha. O eixo x varia de 0 a 20.000 e o eixo y varia de 0 a

Reviewer: ```suggestion
#| O eixo x varia de 0 a 20.000 e o eixo y varia de 0 a
```

#| 5000. As linhas se sobrepõem bastante, sugerindo frequência semelhante nas
#| distribuições dos preços dos diamantes. Uma característica notável é que
#| Os diamantes ideais têm o pico mais alto por volta de 1500.

ggplot(diamante, aes(x = preco)) + 
  geom_freqpoly(aes(color = corte), binwidth = 500, linewidth = 0.75)
```

Note que o ggplot2 usa uma escala de cor ordenada para `corte` porque ele está definido como um fator ordenado nos dados.
Você irá aprender mais sobre isso em @sec-ordered-factors.

A aparência padrão de `geom_freqpoly()` não é tão útil aqui porque a altura, determinada pela contagem geral, difere muito entre os `corte`s, tornando difícil ver as diferenças nas formas de suas distribuições.

> Para fazer a comparação mais fácil nós necessitamos trocar o que está no eixo y.

Reviewer 2: ```suggestion
Para fazer a comparação mais facilmente, precisamos trocar o que está no eixo y.
```

Ao invés da contagem (frequência), nós iremos exibir a **densidade** (**density**), que é a contagem padronizada para que a área sob cada polígono de frequência seja um.

```{r}
#| fig-alt: |
#| Um polígono de frequência de densidades de preços de diamantes onde cada corte de
#| quilate (razoável, bom, muito bom, premium e ideal) é representado por um
#| linha de cor diferente. O eixo x varia de 0 a 20.000. As linhas se sobrepõem
#| bastante, sugerindo distribuições de densidade semelhantes de preços de
#| diamantes. Uma característica notável é que todos os diamantes, exceto os Justos,têm

> #| picos altos em torno de um preço de 1.500 e os diamantes justos têm uma média mais

Reviewer 2: Acho que só usando a letra maiúscula ja ajuda na ideia.

> #| elevada do que outros.

Reviewer: ```suggestion
#| alta do que outros.
```

ggplot(diamante, aes(x = preco, y = after_stat(density))) + 
  geom_freqpoly(aes(color = corte), binwidth = 500, linewidth = 0.75)
```

Note que nós estamos mapeando a densidade para o eixo `y`, mas como `density` não é uma variável do `diamante`, nós precisamos primeiro calcular ela.

> Nós usamos a função `afer_stat()` para fazer isso.

Reviewer 2: ```suggestion
Nós usamos a função `after_stat()` para fazer isso.
```

> Há algo bastante surpreendente neste gráfico - parece que os diamantes razoáveis (de qualidade mais baixa) têm o preço médio mais alto!

Reviewer: ```suggestion
Há algo bastante surpreendente neste gráfico - parece que os diamantes Justos (de qualidade mais baixa) têm o preço médio mais alto!
```

Mas talvez seja porque os polígonos de frequência são um pouco difíceis de interpretar -- há muita coisa acontecendo neste gráfico.
Um gráfico visualmente mais simples para explorar essa relação são os boxplot.

```{r}
#| fig-alt: |

> #| Boxplots lado a lado dos preços dos diamantes por corte. A distribuição de

Reviewer: ```suggestion
#| Boxplots lado a lado dos preços dos diamantes segundo o corte. A distribuição de
```
achei q soava melhor

> #| preços é enviesado à direita para cada corte (razoável, bom, muito bom, premium e

Reviewer: ```suggestion
#| preços é assimétrica à direita para cada corte (Justo,  Bom, Muito Bom, Premium e
```

Isto apoia a descoberta contra-intuitiva de que diamantes de melhor qualidade são normalmente mais baratos!
Nos exercícios, você será desafiado a descobrir o porquê.

`corte` é um fator ordenado: razoável é pior que bom, que por suas vez é pior que muito bom e assim por diante.
Muitas variáveis categóricas não tem uma ordem intrínseca, então você pode querer reordenar elas para tornar uma exibição mais informativa.
Um caminho para isso é usar a `fct_reorder()`.
Você irá aprender mais sobre essa função em @sec-modifying-factor-order, mas nós queremos dar a você uma rápida pré-visualização do uso dela.
Por exemplo, pegue a variável `classe` do banco de dados `milhas`.
Você pode estar interessado em saber como a quilometragem da rodovia varia entre as classes:

```{r}
#| fig-alt: |
#| Boxplots lado a lado de quilometragem rodoviária de carros por classe. As classes
#| no eixo x (2 lugares, compacto, médio, minivan, picape, subcompacto,
#| e SUV).

ggplot(milhas, aes(x = classe, y = rodovia)) +
  geom_boxplot()
```

Para tornar a tendência mais fácil de visualizar, nós podemos reordenar `classe` baseado no valor mediano de `rodovia`:

```{r}
#| fig-alt: |

> #| Boxplots lado a lado de quilometragem rodoviária de carros por classe. As classes

Reviewer: ```suggestion
#| Boxplots lado a lado do consumo de combustível dos carros na rodovia segundo classe. As classes apresentadas 
```

> #| no eixo x estão ordenadas pelo aumento da quilometragem média da rodovia (recolha,

Reviewer: ```suggestion
#| no eixo x estão ordenadas segundo o aumento do consumo mediano de combustível na rodovia (picape,
```

Reviewer 2: Aqui realmente é necessário prestar atenção pois não está falando de average meleage, que seria a quilometragem média. É median, ou seja, a posição, mediana.

#| suv, minivan, 2 lugares, subcompacto, compacto e médio).

ggplot(milhas, aes(x = fct_reorder(classe, rodovia, median), y = rodovia)) +
  geom_boxplot()
```

> Se você tem nomes longos de variáveis, `geom_boxplot()` irá trabalhar melhor se você virar em 90º.

Reviewer: ```suggestion
Se você tem nomes variáveis com nomes longos, `geom_boxplot()` irá funcionar melhor se você girá-los em 90º.
```
não sei se gostei de "girar" tampouco  hehe talvez rotacionar?

Reviewer 2: Acho que está ok. Parece a necessidade do povo em trocar "usar" por "utilizar" quando tem exatamente o mesmo significado mas sentem que "usar" é mais vulgar.

Você pode fazer isso trocando os eixos x e y.

```{r}
#| fig-alt: |

> #| Boxplots lado a lado de quilometragem rodoviária de carros por classe. As classes

Reviewer: ```suggestion
#| Boxplots lado a lado do consumo de combustível dos carros na rodovia segundo classe. As classes
```

> #| estão ordenadas no eixo y pelo aumento da quilometragem média.

Reviewer: ```suggestion
#| estão ordenadas no eixo y segundo o aumento mediano do consumo de combustível.
```

ggplot(milhas, aes(x = rodovia, y = fct_reorder(classe, rodovia, median))) +
  geom_boxplot()
```

#### Exercícios

> 1.  Use o que você tem aprendido para melhorar a visualização dos horários de partida de voos cancelados e não cancelados.

Reviewer: ```suggestion
1.  Use o que você aprendeu para melhorar a visualização dos horários de partida de voos cancelados e não cancelados.
```

2.  Com base na EDA, qual variável do banco de dados diamante parece ser mais importante para predizer o preço de um diamante?
    Como essa variável está correlacionada com corte?

>     Por que a combinação dessas duas relações leva a qualidade mais baixa de diamantes ser mais caros?

Reviewer: ```suggestion
    Por que a combinação dessas duas relações resulta em diamantes de pior qualidade terem preços mais elevados?
```
Liberdade poética para deixar melhor em português

3.  Ao invés de trocar as variáveis x e y, adicione `coord_flip()` como uma nova camada para transformar o boxplot de vertical para horizontal.
    Como isso se compara à troca de variáveis?

> 4.  Um problema com os boxplots é que eles foram desenvolvidos em uma era de conjuntos de dados muito menores e tendem a exibir um número proibitivamente grande de "valores periféricos".

Reviewer: ```suggestion
4.  Um problema com os boxplots é que eles foram desenvolvidos em uma era de conjuntos de dados muito menores e tendem a exibir um número proibitivamente grande de "valores discrepantes".
```

Reviewer 2: De acordo, periférico aqui não faz sentido.

Reviewer 2: Sigo daqui amanhã.

    Uma abordagem para remediar esse problema é o gráfico de valores de letras.

>     Instale o pacote the lvplot, e tente usar `geom_lv()` para exibir a distribuição de preço vs. corte.

Reviewer: ```suggestion
    Instale o pacote lvplot, e tente usar `geom_lv()` para exibir a distribuição de preço vs. corte.
```

    O que você entende?
    Como você interpreta os gráficos?

> 5.  Crie uma visualização dos preços dos diamantes vs. uma variável categórica do banco de dados diamante usando o `geom_violin()`, depois um facetado `geom_histogram()`, e depois um colorido `geom_freqpoly()`, e depois um colorido `geom_density()`.

Reviewer: ```suggestion
5.  Crie uma visualização dos preços dos diamantes vs. uma variável categórica do banco de dados diamante usando o `geom_violin()`, depois um `geom_histogram()` facetado, e depois um `geom_freqpoly()`  colorido, e então um `geom_density()` colorido .
```

    Compare e contraste os quatro gráficos.
    Quais são os prós e contras de cada método de visualização da distribuição de uma variável numérica baseada nos níveis de uma variável categórica?

> 6.  Se você tem um pequeno conjunto de dados, às vezes é útil usar `geom_jitter()` para evitar plotagem excessiva e ver mais facilmente a relação entre as variáveis contínuas e categóricas.

Reviewer: ```suggestion
6.  Se você tem um pequeno conjunto de dados, às vezes é útil usar `geom_jitter()` para evitar sobreposição dos pontos e visualizar mais facilmente a relação entre as variáveis contínuas e categóricas.
```

>     O pacote ggbeeswarm disponibiliza um número de métodos similares ao `geom_jitter()`.

Reviewer: ```suggestion
    O pacote ggbeeswarm disponibiliza vários métodos similares ao `geom_jitter()`.
```

>     Liste eles e descreva brevemente o que cada um deles faz.

Reviewer: ```suggestion
    Liste-os e descreva brevemente o que cada um deles faz.
```

> ### Duas variáveis categóricas

Reviewer: continuar daqui

Para visualizar a covariação entre variáveis categóricas, você irá necessitar contar o número de observações para cada combinação dos níveis dessas variáveis categóricas.
Uma maneira de fazer isso é contar com o `geom_count()` integrado:

```{r}
#| fig-alt: |
#| Um gráfico de dispersão de cor versus corte de diamantes. Há um ponto para cada
#| combinação de níveis de corte (Razoável, Bom, Muito Bom, Premium e Ideal)
#| e cor (D, E, F, G, G, I e J). Os tamanhos dos pontos representam
#| o número de observações para essa combinação. A legenda indica
#| que esses tamanhos variam entre 1000 e 4000.

ggplot(diamante, aes(x = corte, y = cor)) +
  geom_count()
```

O tamanho de cada círculo no gráfico exibe quantas observações ocorrem em cada combinação de valores.
A covariação aparecerá como uma forte correlação entre valores específicos de x e valores específicos de y.

> Outra abordagem para explorar a relação entre essas variáveis é calcular as contagens com dplyr:

Reviewer 2: ```suggestion
Outra abordagem para explorar a relação entre essas variáveis é calcular as contagens com o dplyr:
```

menores_diamantes <- diamante |> 
  filter(quilate < 3)

ggplot(menores_diamantes, aes(x = quilate, y = preco)) +
  geom_point()
```

(Nesta seção usaremos o conjunto de dados `menores_diamantes` para manter o foco na maior parte dos diamantes menores que 3 quilates)

Gráficos de dispersão se tornam menos usuais à medida que seu conjunto de dados cresce, porque os pontos se acumulam, e se amontoam em áreas de preto uniforme, tornando difícil julgar diferenças na densidade dos dados em torno do espaço bidimensional, bem como difícil de ver alguma tendência.
Você já tem visto uma forma de corrigir esse problema: usando a estética (aesthetic) `alpha` para adicionar transparência.

```{r}
#| dev: "png"
#| fig-alt: |
#| Um gráfico de dispersão de preço versus quilate. A relação é positiva, um tanto
#| forte e exponencial. Os pontos são transparentes, mostrando agrupamentos onde
#| o número de pontos é maior do que outras áreas. Os agrupamentos mais óbvios
#| são para diamantes com 1, 1,5 e 2 quilates.

ggplot(menores_diamantes, aes(x = quilate, y = preco)) + 
  geom_point(alpha = 1 / 100)
```

Mas usar transparência pode ser desafiador para conjunto de dados grandes.

> Outra solução é usar intervalos (bins).

Reviewer 2: ```suggestion
Outra solução é usar classes (bins).
```

Previamente, você usou `geom_histogram()` e `geom_freqpoly()` para bin em uma dimensão.
Agora você irá aprender como usar `geom_bin2d()` e `geom_hex()` para bins em duas dimensões.

`geom_bin2d()` e `geom_hex()` divide o plano de coordenadas em bins 2D e então usa um preenchimento de cor para exibir quantos pontos caem em cada bin.
`geom_bin2d()` cria bins retangulares e `geom_hex()` cria bins hexagonais.
Você irá necessitar instalar o pacote hexbin para usar `geom_hex()`.

```{r}
#| layout-ncol: 2
#| fig-width: 3
#| fig-alt: |
#| Plot 1: Um gráfico de densidade de preço versus quilate. Plot 2: Uma caixa hexagonal
#| gráfico de preço versus quilate. Ambos os gráficos mostram que a maior densidade de
#| os diamantes têm quilates e preços baixos.

ggplot(menores_diamantes, aes(x = quilate, y = preco)) +
  geom_bin2d()

# install.packages("hexbin")
ggplot(menores_diamantes, aes(x = quilate, y = preco)) +
  geom_hex()
```

> Outra opção é agrupar uma variável contínua para que ela atue como uma variável categórica.

Reviewer: ```suggestion
Outra opção é discretizar uma variável contínua para que ela atue como uma variável categórica.
```

> Então você usa uma das técnicas para visualizar a combinação de uma variável categórica e uma contínua que você aprendeu antes.

Reviewer: ```suggestion
Assim você consegue usar uma das técnicas para visualizar a combinação de uma variável categórica e uma contínua que você aprendeu antes.
```

Por exemplo, você poderia categorizar a variável `quilate`, e então para cada grupo exibir um boxplot:

```{r}
#| fig-alt: |

> #| Gráficos boxplot lado a lado de preços por quilate. Cada box plot representa

Reviewer: ```suggestion
#| Gráficos boxplot lado a lado de preços por quilate. Cada boxplot representa
```

#| diamantes que têm 0,1 quilates de diferença em peso. Os gráficos de caixa mostram
#| que à medida que quilate aumenta ocorrem aumentos do preço médio. Além disso,

> #| diamantes com 1,5 quilates ou menos têm distribuições de preços distorcidas para a

Reviewer: ```suggestion
#| diamantes com 1,5 quilates ou menos têm distribuições de preços assimétricos à
```

#| direita, 1,5 a 2 têm distribuições de preços aproximadamente simétricas e os
#| diamantes que pesam mais têm distribuições distorcidas à esquerda. Diamantes menores
#| e mais baratos têm valores discrepantes na extremidade superior, diamantes maiores,
#| e mais caros têm valores discrepantes na extremidade inferior.

ggplot(menores_diamantes, aes(x = quilate, y = preco)) + 
  geom_boxplot(aes(group = cut_width(quilate, 0.1)))
```

`cut_width(x, width)`, as used above, divides `x` into bins of width `width`.
By default, boxplots look roughly the same (apart from number of outliers) regardless of how many observations there are, so it's difficult to tell that each boxplot summaries a different number of points.

> One way to show that is to make the width of the boxplot proportional to the number of points with `varwidth = TRUE`.

Reviewer 2: O mesmo vale para essa parte. Remover.

Depois, exponenciamos os resíduos para colocá-los de volta na escala de preços brutos.

```{r}
#| message: false
#| dev: "png"
#| fig-alt: |
#| Um gráfico de dispersão de resíduos versus quilate de diamantes. O eixo x varia de 0
#| a 5, o eixo y varia de 0 a quase 4. Muitos dos dados estão agrupados
#| em torno de valores baixos de quilate e resíduos. Há um padrão claro e curvo
#| mostrando diminuição nos resíduos à medida que o quilate aumenta.

library(tidymodels)

diamante <- diamante |>
  mutate(
    log_preco = log(preco),
    log_quilate = log(quilate)
  )

diamante_fit <- linear_reg() |>
  fit(log_preco ~ log_quilate, data = diamante)

diamante_aug <- augment(diamante_fit, new_data = diamante) |>
  mutate(.resid = exp(.resid))

ggplot(diamante_aug, aes(x = quilate, y = .resid)) + 
  geom_point()
```

Uma vez que você tem removido a forte relação entre quilate e preço, você pode ver o esperado da relação entre corte e preço: relativo ao tamanho deles, diamantes de melhor qualidade (corte) são mais caros.

```{r}
#| fig-alt: |

> #| Gráficos boxplot lado a lado de resíduos por corte. O eixo x exibe os vários

Reviewer: ```suggestion
#| Gráficos boxplot lado a lado dos resíduos por corte. O eixo x exibe os vários
```

#| cortes (justo para ideal), o eixo y varia de 0 a quase 5. As medianas são
#| bastante semelhantes, entre cerca de 0,75 a 1,25. Cada uma das distribuições 
#| dos resíduos são distorcidos para a direita, com muitos valores discrepantes na
#| extremidade superior.

ggplot(diamante_aug, aes(x = corte, y = .resid)) + 
  geom_boxplot()
```

> Nós não discutimos modelagem nesse livro porque entender o que são os modelos e como eles trabalham é mais fácil uma vez que você tem ferramentas de organização de dados (data wrangling) e programação em mãos.

Reviewer: ```suggestion
Nós não discutiremos modelagem nesse livro porque será mais fácil entender o que são os modelos e como eles funcionam uma vez que você tenha ferramentas de tratamento de dados (*data wrangling*) e de programação em mãos.
```
achei q fluía melhor em português nessa ordem

Reviewer 2: Eu formataria data wrangling em itálico.

## Resumo

> Nesse capítulo você tem aprendido uma variedade de ferramentas para ajudar você a entender a variação dentro dos dados.

Reviewer: ```suggestion
Nesse capítulo você aprendeu uma variedade de ferramentas para ajudar você a entender a variação dentro dos dados.
```

> Você tem visto técnicas que trabalham com uma única variável no tempo e com um par de variáveis.

Reviewer: ```suggestion
Você viu técnicas que trabalham com uma única variável no tempo e com um par de variáveis.
```

Isso pode parecer dolorosamente restritivo se você tem dezenas ou centenas de variáveis nos seus dados, mas elas são a base sobre a qual todas as outras técnicas são construídas.

> No próximo capítulo, nós focaremos sobre ferramentas que podemos usar para comunicar nossos resultados.

Reviewer: ```suggestion
No próximo capítulo, nós focaremos em ferramentas que podemos usar para comunicar nossos resultados.
```
pensei em trocar "comunicar" por "apresentar"

Reviewer 2: E faz sentido, já que a seção do livro é de Comunicação.
